test/queue_list_test.dart

```diff
@@ -258,6 +258,25 @@ void main() {
     expect(queue.cast<Pattern>(), same(queue));
   }, skip: isDart2 ? false : 'Requires a Dart2 runtime');
 
+  test("cast does not throw on mutation when the type is valid", () {
+    var patternQueue = new QueueList<Pattern>()..addAll(['a', 'b']);
+    QueueList<Object> untypedQueue = patternQueue;
+    var stringQueue = untypedQueue.cast<String>();
+    stringQueue.addAll(['c', 'd']);
+
+    expect(stringQueue, ['a', 'b', 'c', 'd']);
```

Comment:
Check that `patternQueue` was also modified.

Reply:
Done.